Create Users Controller

Dbz opened the pull request:
- Updated user specs
- Created Users controller
- Annotated

Dbz committed:
Updated user model
Updated user specs
Created Users controller
Annotated

isisAnchalee commented on app/controllers/users/users_controller.rb:
```diff
@@ -0,0 +1,30 @@
+      if @user.update_attributes(user_params)
+        render :show
+      else
+        render :update
```
render update?

isisAnchalee commented on app/controllers/users/users_controller.rb:
```diff
@@ -0,0 +1,30 @@
+class Users::UsersController < ApplicationController
```
this file should be app/controllers/api/users_controller.rb
and rendering json views????

Dbz committed:
Moved users controller to api
changed name to first_name and last_name
added is_company property to
Updated specs

isisAnchalee commented on app/controllers/api/users_controller.rb:
```diff
@@ -0,0 +1,32 @@
+          render json: @user.errors.full_messages
+        end
+      end
+      render :edit
```
don't need this
